src/OpenLoco/Audio/VehicleChannel.cpp

```diff
@@ -1,18 +1,93 @@
 #include "VehicleChannel.h"
 #include "../Entities/EntityManager.h"
 #include "../Interop/Interop.hpp"
+#include "../Map/TileManager.h"
+#include "../Ui/WindowManager.h"
 #include "../Vehicles/Vehicle.h"
 
 using namespace OpenLoco::Interop;
 
 namespace OpenLoco::Audio
 {
-    static std::pair<SoundId, Channel::Attributes> sub_48A590(const Vehicles::Vehicle2or6* v)
+    // Vehicle Volume is multiplied by the modifier then divided by 8 to get in terms of hundredth decibels
+    // 255 represents full volume
+    // 0 represents no volume
+    constexpr int8_t kVolumeModifierZoomIncrement = -35; // 13.7% decrease in volume for each zoom increment (up to 2)
+    constexpr int8_t kVolumeModifierUnderground = -28;   // 11.0% decrease in volume when underground
+    constexpr uint8_t kVolumeModifierMax = 255;
+
+    constexpr int32_t kVolumeMin = -100'00; // hundredth decibels (-100dB)
+    // Calculated min is 255*255/8=8128 but original has done 256*256/8=8192
+    // We have kept original value but could be changed to correctly represent the full range of volume
+    constexpr int32_t kVehicleVolumeCalcMin = -81'91; // hundredth decibels (-81.91dB)
+
+    constexpr int32_t kPanFalloffStart = 2048;
+    constexpr int32_t kPanFalloffEnd = 3072;
+
+    static int8_t getZoomVolumeModifier(uint8_t zoom)
+    {
+        return std::min<uint8_t>(zoom, 2) * kVolumeModifierZoomIncrement;
+    }
+
+    static int8_t getUndergroundVolumeModifier(const Map::Pos3& pos)
+    {
+        if (pos.x != Location::null && Map::validCoords(pos))
+        {
+            return 0;
+        }
+        auto* surface = Map::TileManager::get(pos).surface();
+        if (surface->baseZ() * 4 > pos.z)
+        {
+            return kVolumeModifierUnderground;
+        }
+        return 0;
+    }
+
+    static uint8_t getFalloffModifier(int32_t pan)
     {
-        registers regs;
-        regs.esi = X86Pointer(v);
-        call(0x0048A590, regs);
-        return { static_cast<SoundId>(regs.eax), { regs.ecx, regs.edx, regs.ebx } };
+        const auto absPan = std::min(std::abs(pan), 4095); // This is pointless subsequent if's take care of max
```

Comment:
what does 4095 represent, can it be a constant too?

Reply:
well the problem is is that its a pointless constant it doesn't do anything... I might just delete it.